src/core/gui/menus/popoverMenus/PageTypeSelectionPopover.cpp

```diff
@@ -105,90 +127,308 @@ GtkWidget* createPreviewGrid(const std::vector<std::unique_ptr<PageTypeInfo>>& p
+PageTypeSelectionPopover::PageTypeSelectionPopover(Control* control, Settings* settings, GtkApplicationWindow* win):
+        PageTypeSelectionMenuBase(control->getPageTypes(), settings, PAGETYPE_SELECTION_ACTION_NAME),
+        control(control),
+        controller(control->getPageBackgroundChangeController()),
+        settings(settings),
+        selectedPageSize(getInitiallySelectedPaperSize(settings)),
+        selectedOrientation(selectedPageSize ? selectedPageSize->orientation() : PaperOrientation::VERTICAL) {
     static_assert(is_action_namespace_match<decltype(win)>(G_ACTION_NAMESPACE));
 
+    PaperFormatUtils::loadDefaultPaperSizes(paperSizeMenuOptions);
+    // Add Special options
+    customPaperSizeIndex = paperSizeMenuOptions.size();
+    paperSizeMenuOptions.emplace_back(_("Custom"));
+    copyCurrentPaperSizeIndex = paperSizeMenuOptions.size();
+    paperSizeMenuOptions.emplace_back(_("Copy current page"));
+
+    this->controller->setPaperSizeForNewPages(this->selectedPageSize);
+
+    orientationAction.reset(createOrientationGAction(selectedOrientation), xoj::util::adopt);
+    g_action_map_add_action(G_ACTION_MAP(win), G_ACTION(orientationAction.get()));
+    g_simple_action_set_enabled(orientationAction.get(), selectedPageSize.has_value());
+    g_signal_connect(G_OBJECT(orientationAction.get()), "change-state", G_CALLBACK(changedOrientationSelectionCallback),
+                     this);
+
     g_action_map_add_action(G_ACTION_MAP(win), G_ACTION(typeSelectionAction.get()));
 
-    controller->setPageTypeForNewPages(this->selectedPT);
+    comboBoxChangeSelectionAction.reset(createPageSizeComboBoxChangeSelectionGAction(), xoj::util::adopt);
+    g_action_map_add_action(G_ACTION_MAP(win), G_ACTION(comboBoxChangeSelectionAction.get()));
+
+    pageSizeChangedAction.reset(createPageSizeChangedGAction(), xoj::util::adopt);
+    g_action_map_add_action(G_ACTION_MAP(win), G_ACTION(pageSizeChangedAction.get()));
+}
+template <bool changeComboBoxSelection>
+void PageTypeSelectionPopover::setSelectedPaperSize(const std::optional<PaperSize>& newPageSize) {
+    if (newPageSize != selectedPageSize) {
+        if constexpr (changeComboBoxSelection) {
+            ignoreComboBoxSelectionChange = true;
+            g_action_activate(G_ACTION(comboBoxChangeSelectionAction.get()),
+                              g_variant_new_uint32(getComboBoxIndexForPaperSize(newPageSize)));
+            ignoreComboBoxSelectionChange = false;
+        }
+
+        selectedPageSize = newPageSize;
+        if (selectedPageSize) {
+            g_simple_action_set_state(orientationAction.get(),
+                                      g_variant_new_boolean(static_cast<bool>(selectedPageSize->orientation())));
+        }
+        g_simple_action_set_enabled(orientationAction.get(), selectedPageSize.has_value());
+
+        // Updates sensitivity of 'apply to current'-button
+        g_action_activate(G_ACTION(this->pageSizeChangedAction.get()), nullptr);
```

Comment:
I would have very much liked to not use a GAction there. But to update the sensitivity I would need pointers to the button widgets and because the createPopover method, in which these widgets are created, has to be const I can not save the pointers from there . The only solution I could think of yet was using GActions which activate a callback added in the createPopover method, interacting with the widgets, but I would be happy to find a simpler solution for this